charts/theia.cloud/templates/instances-ingress.yaml

```
@@ -1,33 +1,29 @@
+{{- if not .Values.hosts.usePaths }}
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   name: {{ tpl (.Values.ingress.instanceName | toString) . }}
   namespace: {{ .Release.Namespace }}
   annotations:
     kubernetes.io/ingress.class: nginx
-    {{- if not .Values.hosts.usePaths }}
+    {{- if not .Values.hosts.tls }}
+    nginx.ingress.kubernetes.io/ssl-redirect: "false"
+    {{- end }}
     cert-manager.io/cluster-issuer: {{ tpl (.Values.ingress.clusterIssuer | toString) . }}
-    {{ if .Values.ingress.theiaCloudCommonName }}cert-manager.io/common-name: "Theia.Cloud" {{ end }}
+    {{- if .Values.ingress.theiaCloudCommonName }}cert-manager.io/common-name: "Theia.Cloud" {{- end }}

```

The added `-` at the beginning of the `if` breaks the template:
The `-` removes all whitespace - including line breaks (!) - until it hits the first non-whitespace character. This results in the annotation moving up into the preceding line. Thus the beginning `-` is only used if there is a line break right after the if.

The added `-` in for the `end` is nice though as it avoids trailing whitespace at the end of the line.

That being said, I'd just go for the multi line solution here, too, while we're already changing it. Makes it easier to read IMO.

```suggestion
    {{- if .Values.ingress.theiaCloudCommonName }}
    cert-manager.io/common-name: "Theia.Cloud"
    {{- end }}
```